service/src/main/java/bio/terra/profile/service/iam/SamService.java

```diff
@@ -110,6 +107,28 @@ public boolean isAuthorized(
     }
   }
 
+  public void verifyResourceAdmin(AuthenticatedUserRequest user, SamAction action)
+      throws InterruptedException {
+    String accessToken = user.getToken();
+    var adminApi = samAdminApi(accessToken);
+    try {
+      var isAuthorized =
+          SamRetry.retry(
+              () ->
+                  adminApi.resourceTypeAdminPermission(
+                      SamResourceType.RESOURCE_ADMIN.getSamResourceName(),
```

Comment:
You are correct. There was a conversation I had with Doug that made me think otherwise, but I managed to actually try it out in dev while impersonating the rawls service user.